Bot.DSharp/DSharpClient.cs

```diff
@@ -40,19 +41,38 @@ public async Task ConnectAsync(IServiceProvider botServices)
         {
             var slash = m_discord.UseSlashCommands(new SlashCommandsConfiguration { Services = botServices });
 
-            // During development, register our commands to the dev guild only
-            List<ulong> devGuildIds = new() { 128585855097896963ul, 215551375608643586ul };
+            // If somehow neither deploy type is specified, don't screw around with anything
+            string deployType = Environment.GetEnvironmentVariable("DEPLOY_TYPE") ?? "none";
 
-            foreach (ulong devGuildId in devGuildIds)
-            { 
-                slash.RegisterCommands<DSharpGameSlashCommands>(devGuildId);
-                slash.RegisterCommands<DSharpMessagingSlashCommands>(devGuildId);
-                slash.RegisterCommands<DSharpLookupSlashCommands>(devGuildId);
-                slash.RegisterCommands<DSharpMiscSlashCommands>(devGuildId);
-                slash.RegisterCommands<DSharpSetupSlashCommands>(devGuildId);
+            // The "dev" deploy should only be used by us doing development locally and using the 
+            // DEV bot token - so the DEV bot will remain registered only to our servers, with no global
+            // commands.
+            // The "prod" deploy should only be used by the real server using the real bot token,
+            // so it will now be registered globally
+            if (deployType == "dev")
+            {
+                // During development, register our commands to the dev guild only
+                List<ulong> devGuildIds = new() { 128585855097896963ul, 215551375608643586ul };
+
+                foreach (ulong devGuildId in devGuildIds)
+                {
+                    slash.RegisterCommands<DSharpGameSlashCommands>(devGuildId);
```

Comment:
If we add new commands, this has potential to be problematic because we have to repeat ourselves. I could see us accidentally adding dev commands up here but forgetting to add them to prod below.

I would put the list of types we want to register in a single list, then call either a dev or prod method that loops through the list and registers them all.

Reply:
So like:
```
var allCommands = new[]
{
    typeof(DSharpMessagingSlashCommands),
    typeof(DSharpLookupSlashCommands),
    ...
};

if (dev)
{
  foreach guild
     RegisterCommands(allCommands, guildId);
  slash.RegisterCommands<EmptyCommands>();
}
else if (prod)
{
   RegisterCommands(allCommands);
}
```

Reply:
good idea